CMakeLists.txt

```diff
@@ -82,6 +82,12 @@ if(CMAKE_C_FLAGS_RELEASE MATCHES "-O3")
   string(REPLACE "-O3" "-O2" CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE}")
 endif()
 
+# Enable link-time optimisations on release builds.
+check_c_compiler_flag(-flto HAS_FLTO_FLAG)
+if(HAS_FLTO_FLAG)
+  set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS} -flto")
```

Comment:
Maybe it's only needed for gcc? I (and probably most OSX users?) usually build using clang.

For my quick test I simply added `-lfto` to this line: https://github.com/neovim/neovim/blob/master/CMakeLists.txt#L144